### PR TITLE
(CDAP-6654) Remove dependency on test-jar from cdap-master

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

**Comment by anwar6953** [comment]
lgtm

File: cdap-master/pom.xml [hunk 1]
```diff
@@ -58,12 +58,6 @@
     </dependency>
     <dependency>
       <groupId>co.cask.cdap</groupId>
-      <artifactId>cdap-common</artifactId>
-      <version>${project.version}</version>
-      <type>test-jar</type>
-    </dependency>
```
Comment (gsps1) [inline cdap-master/pom.xml:64]: did we run master on cluster with this change, this shouldn't hopefully break starting master.

File: cdap-master/src/main/java/co/cask/cdap/data/tools/DataMigration.java [hunk 2]
```diff
@@ -220,4 +228,46 @@ public static DatasetFramework createRegisteredDatasetFramework(Injector injecto
...
+    public Location get(NamespaceMeta namespaceMeta) throws IOException {
+      return get(namespaceMeta.getNamespaceId().toId(), null);
+    }
+
+    @Override
+    public Location get(Id.Namespace namespaceId, @Nullable String subPath) throws IOException {
+      // Only allow operates on system namespace
+      if (!Id.Namespace.SYSTEM.equals(namespaceId)) {
+        throw new IllegalArgumentException("Location operation on namespace " + namespaceId +
+                                             " is not allowed. Only SYSTEM namespace is support.");
```
Comment (gsps1) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/DataMigration.java]: `supported`

File: cdap-master/pom.xml (hunk 1 above)
Comment (chtyim) [inline cdap-master/pom.xml:64]: yes

File: cdap-master/src/main/java/co/cask/cdap/data/tools/DataMigration.java (hunk 2 above)
Comment (chtyim) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/DataMigration.java]: fixed

**Comment by chtyim** [comment]
This change doesn't affect unit-test and the checkstyle build passed.

http://builds.cask.co/browse/CDAP-DRC4188

**Comment by chtyim** [comment]
Squashed and merging.

**Pull request merged**